pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -1410,6 +1467,198 @@ func compareCAPIInfraMachines(platform configv1.PlatformType, infraMachine1, inf
 	}
 }
 
+// ensureCAPIMachineStatusUpdated updates the CAPI machine status if changes are detected and conditions are met.
+func (r *MachineSyncReconciler) ensureCAPIMachineStatusUpdated(ctx context.Context, mapiMachine *machinev1beta1.Machine, existingCAPIMachine *clusterv1.Machine, convertedCAPIMachine *clusterv1.Machine, updatedOrCreatedCAPIMachine *clusterv1.Machine, capiMachinesDiff map[string]any, specUpdated bool) (bool, error) {
+	logger := log.FromContext(ctx)
+
+	// If there are no status changes and the spec has not been updated, return early.
+	if !hasStatusChanges(capiMachinesDiff) && !specUpdated {
```

Comment:
One thing coming to my mind:

I think `hasStatusChanges` will always be true, because our status conditions hever have a lastTransitionTime.
This is okay because we check if we really have a diff later on, but this check here is basically useless due to that.

Reply:
Note: the above is not true due to the way we still diff: https://github.com/openshift/cluster-capi-operator/pull/365#discussion_r2379395225